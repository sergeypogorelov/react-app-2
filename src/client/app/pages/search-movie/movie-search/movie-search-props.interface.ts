export interface MovieSearchProps {
  search?: string;
  controlLabeledBy?: string[];
  onSearchFormSubmit?: (searchValue: string) => void;
  onSearchValueChange?: (searchValue: string) => void;
}
