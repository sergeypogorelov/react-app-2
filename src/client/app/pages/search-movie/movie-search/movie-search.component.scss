@import '../../../../styles/variables';

.movie-search {
  &-control {
    width: 700px;
    height: 60px;
    padding-left: 20px;
    color: $c-default;
    font-size: 20px;
    background-color: rgba(50, 50, 50, 0.8);
    border-width: 0;
    border-radius: 4px;
    // box-shadow: 0 4px 0 0 rgba(246, 82, 97, 0.65);
  }
  &-button {
    width: 220px;
    height: 60px;
    margin-left: 10px;
    font-family: $f-medium;
    font-size: 20px;
    color: $c-default;
    background-color: $c-primary;
    border-width: 0;
    border-radius: 4px;
  }
}
