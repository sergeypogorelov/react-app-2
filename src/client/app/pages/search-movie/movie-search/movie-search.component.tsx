import './movie-search.component.scss';

import React, { FunctionComponent } from 'react';

import { MovieSearchProps } from './movie-search-props.interface';

export const MovieSearch: FunctionComponent<MovieSearchProps> = ({
  controlLabeledBy,
}) => {
  const ariaLabeledBy = controlLabeledBy ? controlLabeledBy.join(' ') : null;

  return (
    <form className="movie-search">
      <input className="movie-search-control" aria-labelledby={ariaLabeledBy} />
      <button className="movie-search-button">Search</button>
    </form>
  );
};
