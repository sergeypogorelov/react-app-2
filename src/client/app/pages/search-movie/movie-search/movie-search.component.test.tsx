import React from 'react';
import renderer from 'react-test-renderer';

import { MovieSearch } from './movie-search.component';
import { MovieSearchProps } from './movie-search-props.interface';

function createTree(props?: Partial<MovieSearchProps>) {
  if (!props) {
    props = {};
  }

  return renderer.create(<MovieSearch {...props} />);
}

describe('MovieSearch component', () => {
  it('should render correctly if unlabelled', () => {
    const tree = createTree().toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly if labelled', () => {
    const tree = createTree({
      controlLabeledBy: ['controlId'],
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
