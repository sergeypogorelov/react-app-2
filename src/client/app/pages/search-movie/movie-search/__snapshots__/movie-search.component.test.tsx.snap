// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSearch component should render correctly if labelled 1`] = `
<form
  className="movie-search"
  onSubmit={[Function]}
>
  <input
    aria-labelledby="controlId"
    className="movie-search-control"
    onChange={[Function]}
  />
  <button
    className="movie-search-button"
  >
    Search
  </button>
</form>
`;

exports[`MovieSearch component should render correctly if unlabelled 1`] = `
<form
  className="movie-search"
  onSubmit={[Function]}
>
  <input
    aria-labelledby={null}
    className="movie-search-control"
    onChange={[Function]}
  />
  <button
    className="movie-search-button"
  >
    Search
  </button>
</form>
`;
